rascal-vscode-extension/src/util/PositionConverter.ts

```diff
@@ -0,0 +1,189 @@
+import * as vscode from "vscode";
+import { SourceLocation } from '../RascalTerminalLinkProvider';
+
+/**
+ * This class allows to convert text locations between VS Code and Rascal.
+ * Some characters, which VS Code counts as two characters because it uses UTF-16 encoding,
+ * are only a single character for Rascal, which uses UTF-32 encoding.
+ *
+ * Assuming UTF-16 encoding, because this code is executed in TypeScript,
+ * all methods in this class therefore iterate character-wise over a text to find pairs of characters
+ * that would be encoded as a single character in UTF-32.
+ *
+ * If they find such a pair, they change the position to encount for the difference in encoding:
+ * If the position comes from Rascal, it is too short and must be increased by 1.
+ * If the position comes from VS Code, it is too long and must be decreased by 1.
+ */
+export class PositionConverter {
+
+    /***************************************
+     *              Rascal -> VS Code      *
+     ***************************************/
+
+    /**
+     * Converts the column position from Rascal (UTF-32) to VS Code (UTF-16).
+     * @param td the text document where the column information is located in.
+     * @param line the line in which the column to be changed is located.
+     * @param rascalColumn the column as given by Rascal.
+     * @returns the column as understood by VS Code.
+     */
+    static rascalToVSCodeColumn(td: vscode.TextDocument, line: number, rascalColumn: number): number {
+        const fullLine = td.lineAt(line).text;
```

Comment:
shouldn't this be `line - 1`? (and maybe named `rascalLine`?) Or should we document that we assume the caller already has converted the line number to vs code?